src/fsharp/NameResolution.fs

```diff
@@ -2148,7 +2148,7 @@ let rec ResolveFieldInModuleOrNamespace (ncenv:NameResolver) nenv ad (resInfo:Re
             match TryFindTypeWithRecdField modref id  with
             | Some tycon when IsEntityAccessible ncenv.amap m ad (modref.NestedTyconRef tycon) -> 
                 let showDeprecated = HasFSharpAttribute ncenv.g ncenv.g.attrib_RequireQualifiedAccessAttribute tycon.Attribs
-                success(FieldResolution(modref.RecdFieldRefInNestedTycon tycon id,showDeprecated), rest)
+                success(resInfo, FieldResolution(modref.RecdFieldRefInNestedTycon tycon id,showDeprecated), rest)
```

Comment:
This is including some changes where field resolution in FCS tracks the module-path resolution to the fields, just like we do for all other kinds of identifiers.  FCS does this to report the module symbols in ``Module`.Module2.field3``.   However this commit doesn't include the actual call to the sink to report the symbols, that call will be added in a separate commit when we can make sure we also have testing for that.